cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/RenameController.java

```diff
@@ -0,0 +1,87 @@
+package cloud.foundry.cli.services;
+
+import static picocli.CommandLine.*;
+import static picocli.CommandLine.usage;
+
+import cloud.foundry.cli.crosscutting.exceptions.UpdateException;
+import cloud.foundry.cli.crosscutting.logging.Log;
+import cloud.foundry.cli.crosscutting.mapping.CfOperationsCreator;
+import cloud.foundry.cli.operations.ApplicationsOperations;
+import cloud.foundry.cli.operations.ServicesOperations;
+import org.cloudfoundry.operations.DefaultCloudFoundryOperations;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.Callable;
+
+/**
+ * This class realizes the functionality that is needed for the rename commands.
+ * You can rename applications or services.
+ */
+@Command(name = "rename", header = "%n@|green Rename an application or a service.|@",
+        subcommands = {RenameController.RenameApplicationCommand.class,
+        RenameController.RenameServiceCommand.class})
+public class RenameController implements Callable<Integer> {
+
+    @Override
+    public Integer call() throws Exception {
+        usage(this, System.out);
+        return 0;
+    }
+
+    @Command(name = "application", description = "Rename an application.")
+    static class RenameApplicationCommand implements Callable<Integer> {
+
+        private static final Log log = Log.getLog(RenameApplicationCommand.class);
+
+        @Mixin
+        LoginCommandOptions loginOptions;
```

Comment:
You should be able to set this field to private too

Reply:
done